test/app.e2e-spec.ts

```diff
@@ -66,6 +68,44 @@ describe('AppController (e2e)', () => {
     return request(app.getHttpServer()).get('/users').expect(200);
   });
 
+  it('Get a Rate limit Error after a number of requests', (done) => {
+    async.series(
+      [
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(200, callback);
+        },
+        function (callback) {
+          request(app.getHttpServer()).get('/users').expect(429, callback);
+        },
+      ],
+      done,
+    );
+  });
```

Comment:
Brainstorming here, but I'm thinking in getting the output for this tests something like this:

![image](https://github.com/Y4j1QTJYfBZBwp0/microservices-nestjs/assets/6560620/2ffb039c-b77a-49de-9185-f547ec16524f)

Couldn't get local running 100% as of yet (need to see what is wrong with my testcontainers setup), but with Jest we can use `it.each`, in such a way that we could do something similar to this:

```typescript
describe.only('Get a Rate limit Error after a number of requests', () => {
  const cases: Array<Array<number>> = [
    [ 1, 200],
    [ 2, 200],
    [ 3, 200],
    [ 4, 200],
    [ 5, 200],
    [ 6, 200],
    [ 7, 200],
    [ 8, 200],
    [ 9, 200],
    [10, 429],
  ];

  function req(_endpoint: string): number {
    req['count'] = req['count'] || 0;
    return ++req['count'] === 10 ? 429 : 200;
  };

  it.each(cases)('request #%i should return %i', (_, expectedStatus) => {
    expect(req('/users')).toEqual(expectedStatus);
  });
});
```

That mock `req` function is just a placeholder as my local is not 100 yet. Just to give an idea anyways.